inc/classes/class-wp-rocket-requirements-check.php

```diff
@@ -0,0 +1,250 @@
+	 *
+	 * @return bool
+	 */
+	private function wp_passes() {
+		global $wp_version;
+
+		return version_compare( $wp_version, $this->wp_version ) >= 0 ? true : false;
+	}
+
+	/**
+	 * Warns if PHP version is less than 5.4 and offers to rollback.
+	 *
+	 * @since 3.0 Updated minimum PHP version to 5.4 and minimum WordPress version to 4.2
+	 * @since 3.0 Moved to class
+	 * @since 2.11
+	 * @author Remy Perona
+	 */
+	public function notice() {
+		/** This filter is documented in inc/admin-bar.php */
+		if ( ! current_user_can( apply_filters( 'rocket_capacity', 'manage_options' ) ) ) {
+			return;
+		}
+
+		// Translators: %1$s = Plugin name, %2$s = Plugin version.
+		$message = '<p>' . sprintf( __( 'To function properly, %1$s %2$s requires at least:', 'rocket' ), $this->plugin_name, $this->plugin_version ) . '</p><ul>';
+
+		if ( ! $this->php_passes() ) {
+			// Translators: %1$s = PHP version required.
+			$message .= '<li>' . sprintf( __( 'PHP %1$s. To use this version, please ask your web host how to upgrade your server to PHP %1$s or higher.', 'rocket' ), $this->php_version ) . '</li>';
+		}
+
+		if ( ! $this->wp_passes() ) {
+			// Translators: %1$s = WordPress version required.
+			$message .= '<li>' . sprintf( __( 'WordPress %1$s. To use this version, please upgrade WordPress to version %1$s or higher.', 'rocket' ), $this->wp_version ) . '</li>';
+		}
+
+		$message .= '</ul><p>If you are not able to upgrade, you can rollback to the previous version by using the button below.</p><p><a href="' . wp_nonce_url( admin_url( 'admin-post.php?action=rocket_rollback' ), 'rocket_rollback' ) . '" class="button">' .
+		// Translators: %s = Previous plugin version.
+		sprintf( __( 'Re-install version %s', 'rocket' ), $this->plugin_last_version )
+		. '</a></p>';
+
+		echo '<div class="notice notice-error">' . $message . '</div>';
+	}
+
+	/**
+	 * Do the rollback
+	 *
+	 * @since 3.0
+	 * @author Remy Perona
+	 */
+	public function rollback() {
+		check_ajax_referer( 'rocket_rollback' );
+
+		$consumer_key = isset( $this->options['consumer_key'] ) ? $this->options['consumer_key'] : false;
+
+		if ( ! $consumer_key && defined( 'WP_ROCKET_KEY' ) ) {
+			$consumer_key = WP_ROCKET_KEY;
+		}
+
+		$plugin_transient = get_site_transient( 'update_plugins' );
+		$plugin_folder    = plugin_basename( dirname( $this->plugin_file ) );
+		$plugin_file      = basename( $this->plugin_file );
+		$url              = sprintf( 'https://wp-rocket.me/%s/wp-rocket_%s.zip', $consumer_key, $this->plugin_last_version );
+		$temp_array       = array(
+			'slug'        => $plugin_folder,
+			'new_version' => $this->plugin_last_version,
+			'url'         => 'https://wp-rocket.me',
+			'package'     => $url,
+		);
+
+		$temp_object = (object) $temp_array;
+		$plugin_transient->response[ $plugin_folder . '/' . $plugin_file ] = $temp_object;
+		set_site_transient( 'update_plugins', $plugin_transient );
+
+		$transient = get_transient( 'rocket_warning_rollback' );
```

Comment:
Just checking you thought about all these things:
- What happens if this transient is set?
- Shouldn't there be a `set_transient()` somewhere?
- Shouldn't be `get_site_transient()` for multisite?
- If the server isn't responding or returns an error, what happens?
(I don't need answers for all of those, just a "Yep, all is handled" is fine ;))

Reply:
I have to say, I did a straight copy/paste of the rollback function we have.

Now, searching in the code, I can see this transient is not set anywhere, so I don't know what's the point, and it might be leftover code from quite some time

Reply:
We keep it as it is for now?